--- Desktop/pve-kernel/submodules/ubuntu-focal/drivers/pci/quirks.c	2020-05-02 10:56:02.000000000 -0400
+++ Desktop/quirks.c	2020-05-02 11:09:26.000000000 -0400
@@ -5040,6 +5040,10 @@
 DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_INTEL, 0x1502, quirk_intel_no_flr);
 DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_INTEL, 0x1503, quirk_intel_no_flr);
 
+/* FLR causes Ryzen 3000s built-in HD Audio & USB Controllers to hang on VFIO passthrough */
+DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_AMD, 0x149c, quirk_intel_no_flr);
+DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_AMD, 0x1487, quirk_intel_no_flr);
+
 static void quirk_intel_th_rtit_bar(struct pci_dev *dev)
 {
 	struct resource *r = &dev->resource[4];
